src/openforms/authentication/contrib/digid_eherkenning_oidc/views.py

```diff
@@ -68,7 +72,18 @@ def failure_url(self):
         """
         f = furl(self.success_url)
         f = furl(f.args["next"])
-        f.args[BACKEND_OUTAGE_RESPONSE_PARAMETER] = self.plugin_identifier
+
+        # When the user cancels login we want an explicit error for DigiD authentication
+        # according to the spec.
+        if (
+            self.request.GET.get("error") == "access_denied"
+            and self.request.GET.get("error_description") == "The user cancelled"
+            and self.plugin_identifier in ["digid_oidc", "digid_machtigen_oidc"]
+        ):
+            f.args[DIGID_MESSAGE_PARAMETER] = LOGIN_CANCELLED
```

Comment:
I think it would be cleaner separation of concerns to implement a `def get_error_message_parameters(self, error: str, error_description: str):` method and implement the DigiD check in the subclasses.

It doesn't sit right with me that subclass-specific behaviour is implemented in the base class, this means you can never alter the behaviour from a third-party plugin because you need to modify the base class behaviour.

Moving this to a custom/helper method improves loose coupling instead of the current tight coupling implementation.